rfcs/0021-glossary-terms.md

```diff
@@ -0,0 +1,123 @@
+
+### Block
+**Concept:** a set of transactions that represent a state transition.
+
+**Rationale:** This term aligns with the rest of the industry, and it makes it easier to onboard other blockchain users.
+
+**Downsides:** Perhaps it is not as technically accurate as a transition.
+
+**Usage:** In the Coda network, transactions get added to blocks.
+
+**Alternatives:**
+- External transition - maybe this is more precise, but has the heavy lifting of having to explain a new term for a mostly similar concept. Additionally, the word transition is very similar to transaction, causing confusion at times.
+
+### User Transaction
+**Concept:** A transaction issued by a user - currently a payment or a delegation change
+
+**Rationale:** This term clearly describes the concept, and leaves room for further types of user issued transactions.
+
+**Downsides:** This is a subset of "Transactions", so it is a bit redundant.
+
+**Usage:** There are three types of transactions in the Coda network currently - user transactions, fee transfers, and coinbases.
+
+**Alternatives:**
+- User command - this term is confusing, especially in  docs, as it can be conflated with a CLI command that a user issues.
+
+### Snark worker / Snarker
+**Concept:** A Coda node that produces SNARKs to compress transactions (and in the future, blocks) in the network.
+
+**Rationale:** The community likes SNARKs and it is a point of differentiation for Coda, so snarker is an engaging term.
+
+**Downsides:** Gets invalidated if Coda switches to another type of zero-knowledge proof. NOTE - There is some concern about using the progressive tense of "snark" as a verb, eg. "snarking" - as this can be confused with staking. However, if the community enjoys "snarking", it  would make sense to  continue using it as a verb.
+
+**Usage:** 
+- Anyone can join the coda network and become a snark worker.
+- Snark workers help compress the blockchain by generating SNARKs.
+
+**Alternatives:**
+- Compressor - this term was considered initially, but given the community excitement about SNARKs, it makes sense to include the more specific term in the lexicon.
+
+### Full node
+**Concept:** A Coda node that is able to verify the state of the network - however it may need to request paths for its accounts from other nodes that have all the accounts state.
+
+**Rationale:** By calling this type of node a full node, Coda distinguishes itself from other networks, as all nodes are technically full nodes if they can verify SNARKs - these nodes are not required to trust other nodes.
+
+**Downsides:** Because these nodes still need to request state from other nodes, there is an argument to be made that they are not full nodes. Furthermore, Bitcoin classifies full nodes as nodes that "download every block and transaction and check them against Bitcoin's consensus rules." Therefore, there may be some community pushback against calling these nodes full nodes, even though they are trustless.
```

Comment:
These downsides make me very hesistant about this position. I understand that calling everything a full node sounds nice, but I don't think it matches with the community's definition of a full node. If we want to go down this road, I would highly encourage sending a poll out to people who are already in our community and seeing what definition of a full node most closely represents their understanding of the phrase and base our decision off of that. If the overwhelming majority think that a full node requires the ability to download every block, then we should really consider a different name.

Reply:
Agreed - I was also apprehensive about this one, but I think it is worth discussing. cc @es92

Reply:
My thinking is I've found "full node" a really good way to explain the advantages of Coda - saying "every node is a full node" seems to communicate it really well to the audiences I've talked to so far, and has fit in with how they think about full nodes without much contention. My thoughts on it are, our nodes are full nodes because it doesn't matter where they ask for the extra data is from, it can be a centralized source, the protocol, wherever, they have the full information with them to be trustless. Its like software that knows the hash of a chunk of data, but the data itself lives on the network and is just a request away (it may not be on-machine, but its just a request away). But I get there's some nuance there.

My thoughts on the tradeoff are (of full node vs not):
- It makes it really clear what the advantages are. If they understand the distinction between full node and "full node that is storing the account database" in some ways that's good, that means they really understand
- I don't want to make anyone upset or feel tricked - I don't think full node does that but if it upsets a particular audience we should take that into account
- I don't have a better word that's not inventing terminology (which causes a higher barrier to entry) or more accurate (doesn't explain the fully-trustless component)

My vote would be to continue with full node, see if it is something people are  authentically taking issue with, and if so change to "all nodes are fully-trustless" or some other alternative.